apm-agent-plugins/apm-jdbc-plugin/src/main/java/co/elastic/apm/agent/jdbc/helper/JdbcHelper.java

```diff
@@ -35,7 +35,12 @@
 public abstract class JdbcHelper {
     @SuppressWarnings("WeakerAccess")
     @VisibleForAdvice
-    public static final WeakConcurrentMap<Object, String> statementSqlMap = new WeakConcurrentMap<>(true);
+    public static final WeakConcurrentMap<Object, String> statementSqlMap;
+
+    static {
+        statementSqlMap = new WeakConcurrentMap<>(true);
+        statementSqlMap.getCleanerThread().setContextClassLoader(null);
```

Comment:
```suggestion
        // This class may be loaded in the context of a request processing thread
        // whose context class loader is the WebApplicationClassLoader.
        // This leaks the WebApplicationClassLoader when the application is undeployed.
        // Tomcat will then stop the thread because it thinks it was created by the web application.
        // That means that the map will never be cleared, creating a severe memory leak.
        statementSqlMap.getCleanerThread().setContextClassLoader(null);
```